src/vaadin-menu-bar-interactions-mixin.html

```diff
@@ -0,0 +1,347 @@
+      if (subMenu && subMenu.opened) {
+        subMenu.close();
+      }
+    }
+
+    _onMouseOver(e) {
+      const button = Array.from(e.composedPath()).filter(el => el.item)[0];
+      if (button && button !== this._expandedButton && (this.openOnHover || (button.item.children && this._subMenu.opened))) {
+        this.__openSubMenu(button);
+      }
+    }
+
+    __onContextMenuKeydown(e) {
+      const item = Array.from(e.composedPath()).filter(el => el._item)[0];
+      if (item) {
+        const list = item.parentNode;
+        if (e.keyCode === 38 && item === list.items[0]) {
+          this._close(true);
+        }
+        // ArrowLeft, or ArrowRight on non-parent submenu item
+        if (e.keyCode === 37 || (e.keyCode === 39 && !item._item.children)) {
+          // Prevent ArrowLeft from being handled in context-menu
+          e.stopImmediatePropagation();
+          this._navigateByKey(e);
+          const button = this.shadowRoot.activeElement;
+          if (button && button.item && button.item.children) {
+            this.__openSubMenu(button, {keepFocus: true});
+          }
+        }
+      }
+    }
+
+    __outsideClickListener(e) {
+      if (!this._getButtonFromEvent(e) && e.composedPath().indexOf(this._subMenu.$.overlay) === -1) {
+        this._close();
+      }
+    }
+
+    __fireItemSelected(value) {
+      this.dispatchEvent(new CustomEvent('item-selected', {detail: {value}}));
+    }
+
+    __onButtonClick(e) {
+      e.stopPropagation();
+      const button = this._getButtonFromEvent(e);
+      if (button) {
+        this.__openSubMenu(button);
+      }
+    }
+
+    __openSubMenu(button, options = {}) {
+      const subMenu = this._subMenu;
+      const item = button.item;
+
+      if (subMenu.opened) {
+        this._close();
+        if (subMenu.listenOn === button) {
+          return;
+        }
+      }
+
+      const items = item && item.children;
+      if (!items || items.length === 0) {
+        this.__fireItemSelected(item);
+        return;
+      }
+
+      subMenu.items = items;
+      subMenu.listenOn = button;
+      this._expandedButton = button;
+
+      const rect = button.getBoundingClientRect();
+
+      requestAnimationFrame(() => {
+        button.dispatchEvent(new CustomEvent('opensubmenu', {detail: {
+          x: rect.left,
+          y: rect.bottom,
+          children: items
+        }}));
+
+        button.setAttribute('expanded', '');
+      });
+
+      if (options.focusLast) {
+        this.__onceOpened(() => this._focusLastItem());
+      }
+
+      if (options.keepFocus) {
+        this.__onceOpened(() => {
+          this._focusButton(this._expandedButton);
+        });
+      }
+    }
+
+    _focusFirstItem() {
+      const list = this._subMenu.$.overlay.firstElementChild;
+      list.focus();
+    }
+
+    _focusLastItem() {
+      const list = this._subMenu.$.overlay.firstElementChild;
+      const item = list.items[list.items.length - 1];
+      item.focus();
```

Comment:
A really special case again, but a falsy check for `item` would avoid an error here, when the user has only components (non-vaadin-item) inside the overlay. I know that at least one Flow user has used context-menu with just a combo-box in the overlay.

Reply:
Thanks, fixed.